src/acp/policy.ts

```diff
@@ -0,0 +1,179 @@
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { acp, rdf } from "../constants";
+import {
+  internal_toIriString,
+  SolidDataset,
+  Thing,
+  ThingPersisted,
+  Url,
+  UrlString,
+} from "../interfaces";
+import { internal_defaultFetchOptions } from "../resource/resource";
+import {
+  createSolidDataset,
+  saveSolidDatasetAt,
+} from "../resource/solidDataset";
+import { getUrl } from "../thing/get";
+import { setUrl } from "../thing/set";
+import {
+  createThing,
+  getThing,
+  getThingAll,
+  isThingLocal,
+  removeThing,
+  setThing,
+} from "../thing/thing";
+
+export type PolicyDataset = SolidDataset;
+export type Policy = ThingPersisted;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new empty [[SolidDataset]] to story [[Policy]]'s in.
+ */
+export const createPolicyDataset = createSolidDataset;
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Mark a given [[SolidDataset]] as containing [[Policy]]'s, and save it to the given URL.
+ *
+ * @param url URL to save this Access Policy SolidDataset at.
+ * @param dataset The SolidDataset containing Access Policies to save.
+ * @param options Optional parameter `options.fetch`: An alternative `fetch` function to make the HTTP request, compatible with the browser-native [fetch API](https://developer.mozilla.org/docs/Web/API/WindowOrWorkerGlobalScope/fetch#parameters).
+ */
+export async function savePolicyDatasetAt(
+  url: Url | UrlString,
+  dataset: PolicyDataset,
+  options: Partial<
+    typeof internal_defaultFetchOptions
+  > = internal_defaultFetchOptions
+): ReturnType<typeof saveSolidDatasetAt> {
+  url = internal_toIriString(url);
+  let datasetThing = getThing(dataset, url) ?? createThing({ url: url });
+  datasetThing = setUrl(datasetThing, rdf.type, acp.AccessPolicyResource);
+  dataset = setThing(dataset, datasetThing);
+
+  return saveSolidDatasetAt(url, dataset, options);
+}
+
+/**
+ * ```{note} There is no Access Control Policies specification yet. As such, this
+ * function is still experimental and subject to change, even in a non-major release.
+ * ```
+ *
+ * Initialise a new, empty [[Policy]].
+ *
+ * @param url URL that identifies this Access Policy.
+ */
+export function createPolicy(url: Url | UrlString): Policy {
+  url = internal_toIriString(url);
+  let policyThing = createThing({ url: url });
+  policyThing = setUrl(policyThing, rdf.type, acp.AccessPolicy);
```

Comment:
It seems that we begin to use this construct quite often. Would it be worth to publicly expose a function `createThingTyped`, which takes the type as a mandatory parameter ? On one hand, we want common operations to be as affordable as possible, but on the other once we go down this path we could have a proliferation of functions that just bake in RDF predicates.

Reply:
Yeah, I was thinking about abstracting out this specific part, but since most of the "code" here is the comments, and how I'd sooner expect this specific function to change separately rather than wanting to update it for everything, I didn't do it for this PR.

That said, I didn't imagine it as a convenience function for developers, which could make sense. I've mentioned it in the ticket about `findThingBy*`, where it might make sense to combine it with a bit of other potential work.

Reply:
Yeah I agree with the convenience function, although I'd call it `createThingOfType(type: iri)`. I wouldn't worry (yet!) about a proliferation of functions that just bake in RDF predicates - `rdf:type` is one of the most common RDF predicates out there, so having convenience functions for it makes a lot of sense (as does a 'getThingOfType()`).